cpp/src/transform/encode.cpp

```diff
@@ -1,14 +1,56 @@
 #include <cudf/column/column.hpp>
 #include <cudf/column/column_view.hpp>
 #include <cudf/copying.hpp>
+#include <cudf/detail/gather.hpp>
 #include <cudf/detail/search.hpp>
+#include <cudf/detail/sorting.hpp>
 #include <cudf/detail/stream_compaction.hpp>
 #include <cudf/detail/transform.hpp>
 #include <cudf/table/table_view.hpp>
+#include <iostream>
+#include <numeric>
 
 namespace cudf {
 namespace detail {
 
+std::pair<std::unique_ptr<table>, std::unique_ptr<column>> encode(
+  table_view const& input_table, rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+{
+  // side effects of this function we are now dependent on:
+  // - resulting table elements are sorted ascending
```

Comment:
Move this comment to where `drop_duplicates` is called.
Or replace 'this function' with 'drop_duplicates'.